test/integration_test.rb

```diff
@@ -4,24 +4,84 @@
 require "test_helper"
 
 class IntegrationTest < Minitest::Test
-  def setup
+  def test_ruby_lsp_doctor_works
     skip("CI only") unless ENV["CI"]
-  end
 
-  def test_ruby_lsp_doctor_works
     in_isolation do
       system("bundle exec ruby-lsp --doctor")
       assert_equal(0, $CHILD_STATUS)
     end
   end
 
   def test_ruby_lsp_check_works
+    skip("CI only") unless ENV["CI"]
+
     in_isolation do
       system("bundle exec ruby-lsp-check")
       assert_equal(0, $CHILD_STATUS)
     end
   end
 
+  def test_adds_bundler_version_as_part_of_exec_command
+    Dir.mktmpdir do |dir|
+      Dir.chdir(dir) do
+        File.write(File.join(dir, "Gemfile"), <<~GEMFILE)
+          source "https://rubygems.org"
+          gem "ruby-lsp", path: "#{Bundler.root}"
+        GEMFILE
+
+        Bundler.with_unbundled_env do
+          capture_subprocess_io do
+            system("bundle install")
+
+            Object.any_instance.expects(:exec).with do |env, command|
+              assert(env.key?("BUNDLE_GEMFILE"))
+              assert(env.key?("BUNDLER_VERSION"))
+              assert_match(/bundle _[\d\.]+_ exec ruby-lsp/, command)
+            end.once.raises(StandardError.new("stop"))
```

Comment:
From the [docs](https://mocha.jamesmead.org/Mocha/Expectation.html#with-instance_method), would this be more idiomatic? Seems it verifies the block returns a truthy value.

```ruby
Object.any_instance.expects(:exec).with do |env, command|
  env.key?("BUNDLE_GEMFILE") &&
  env.key?("BUNDLER_VERSION") &&
  command.mach(/bundle _[\d\.]+_ exec ruby-lsp/)
end
```

Reply:
Good catch. Asserting in the `with` block is not a good practice.